crates/block-producer/src/block_builder/mod.rs

```diff
@@ -268,13 +288,17 @@ impl BlockBuilder {
         body: BlockBody,
         block_proof: BlockProof,
     ) -> Result<ProvenBlock, BuildBlockError> {
+        // SAFETY: The header and body are assumed valid and consistent with the proof.
```

Comment:
This `fn` doesn't need to be `async` anymore. I'm actually surprised `clippy` doesn't complain about this but maybe we've disabled the lint.

Reply:
Would mean I have to wrap it in a future manually in the call chain, probably prefer to keep it in fn signature?
```rust 
.and_then(|(proposed_block, header, body, block_proof)| async {self.construct_proven_block(proposed_block, header, body, block_proof)})
```

Reply:
I've done it similarly in the batch builder 

https://github.com/0xMiden/miden-node/blob/9460b4423f6274441975ced75453b121013cab1b/crates/block-producer/src/batch_builder/mod.rs#L181

Though I also did consider just saying screw it and keeping the async. I think its up to you, at the time I thought its worth indicating to the reader that the function is _not_ async (e.g. it stood out to me on reading just the fn now).